packages/openapi3/test/parameters.test.ts

```diff
@@ -1,33 +1,56 @@
 import { expectDiagnostics } from "@typespec/compiler/testing";
 import { deepStrictEqual, ok, strictEqual } from "assert";
 import { describe, expect, it } from "vitest";
+import { OpenAPI3PathParameter, OpenAPI3QueryParameter } from "../src/types.js";
 import { diagnoseOpenApiFor, openApiFor } from "./test-host.js";
 
-describe("openapi3: parameters", () => {
+describe("query parameters", () => {
+  async function getQueryParam(code: string): Promise<OpenAPI3QueryParameter> {
+    const res = await openApiFor(code);
+    const param = res.paths[`/`].get.parameters[0];
+    strictEqual(param.in, "query");
+    return param;
+  }
+
   it("create a query param", async () => {
-    const res = await openApiFor(
-      `
-      op test(@query arg1: string): void;
+    const param = await getQueryParam(
+      `op test(@query myParam: string): void;
       `
     );
-    strictEqual(res.paths["/"].get.parameters[0].in, "query");
-    strictEqual(res.paths["/"].get.parameters[0].name, "arg1");
-    deepStrictEqual(res.paths["/"].get.parameters[0].schema, { type: "string" });
+    strictEqual(param.name, "myParam");
+    deepStrictEqual(param.schema, { type: "string" });
   });
 
   it("create a query param with a different name", async () => {
-    const res = await openApiFor(
+    const param = await getQueryParam(
```

Comment:
```ts
@route("blobs/{filter*}") op a(@query filter: string[]): void;
```
I was wondering what the path of OpenAPI would look like? Is it `blobs/{filter*}` or `blobs/{filter}`? I believe the latter one according to OpenAPI spec.

Reply:
yes the later one. For the 2nd one (every pr publish a playgorund you can try fyi [playground](https://cadlplayground.z22.web.core.windows.net/prs/3932/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBIdHRwOwoKQHJvdXRlKCIvYmFyP2ZpeGVkPXllc3smZm9vfSIpIG9wIGZvbyhAcXVlcnnECzogc3RyaW5nKTogdm9pZDs%3D&e=%40typespec%2Fopenapi3&options=%7B%7D))  it fails because the openapi3 emitter doesn't like fixed query params(existing error). However if you do something similar with 2 params [playground](https://cadlplayground.z22.web.core.windows.net/prs/3932/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBIdHRwOwoKQHJvdXRlKCIvYmFyez9maXhlZH17JmZvb30iKSBvcCBmb28oCiAgQHF1ZXJ5IMUgOiBzdHJpbmcsyxhvb8oWKTogdm9pZDsK&e=%40typespec%2Fopenapi3&options=%7B%7D)

```
@route("/bar{?fixed}{&foo}") op foo(
  @query fixed: string,
  @query foo: string,
): void;
```

```
/bar
```